File: JS/3498.reverse_degree_of_a_string [1].js
/**
 * @param {string} s
 * @return {number}
 */
var reverseDegree = function(s) {
    let sum = 0
    for(let i=0; i<s.length; i++)
        sum += (26 - (s.charCodeAt(i) - 'a'.charCodeAt(0))) * (i+1)
    
    return sum
};

File: CPP [3001-3500]/3498.reverse_degree_of_a_string [1].cpp
class Solution {
public:
    // T.C.=O(n), S.C.=O(1)
    int reverseDegree(string s) {
        int sum = 0;
        for(int i=0; i<s.length(); i++)
            sum += (26 - (s[i] - 'a')) * (i+1);
        
        return sum;
    }
};